rewrite-yaml/src/main/java/org/openrewrite/yaml/MergeYamlVisitor.java

```diff
@@ -62,14 +64,12 @@ public class MergeYamlVisitor<P> extends YamlVisitor<P> {
     private final String objectIdentifyingProperty;
 
     @Nullable
-    private final String insertBefore;
+    private final InsertMode insertMode;
 
-    private boolean shouldAutoFormat = true;
+    @Nullable
+    private final String insertProperty;
 
-    public MergeYamlVisitor(Yaml.Block block, Yaml incoming, boolean acceptTheirs, @Nullable String objectIdentifyingProperty, boolean shouldAutoFormat) {
-        this(block, incoming, acceptTheirs, objectIdentifyingProperty, null);
-        this.shouldAutoFormat = shouldAutoFormat;
-    }
```

Comment:
Figured I'd call out I saw `MergeYamlVisitor` used directly downstream in rewrite-spring and rewrite-circleci. Be sure to update the usage there after this is merged. Same goes for direct usage of `MergeYaml` itself.